Topics/Tech_Stacks/Bootstrap.md

```diff
@@ -0,0 +1,163 @@
+# Learning Bootstrap, a CSS Framework
+
+## Table of contents
+### [Introduction](#what-is-bootstrap?-1)
+### [What is Bootstrap](#what-is-bootstrap-1)
+### [Where to begin](#where-to-begin-1)
+### [Bootstrap Basics](#bootstrap-basics-1)
+### [References](#references-1)
+
+## Preface
+Here you will learn the basics of how to incorporate Bootstrap into your website-building workflow. This should not serve as an in-depth guide, but rather one that should help you get your foot in the door to begin understanding and using Bootstrap components. Before you start, it is recommended that you have working knowledge of both HTML and CSS. This could be as little as a simple styled website. Although not required, it will be helpful in understanding why Bootstrap is such a widely-used and useful framework when working with CSS. At the end of this guide, you should be able to apply some simple Bootstrap components or formatting into your website.
+
+## What is Bootstrap?
+Bootstrap is a CSS Framework that simplifies and standardizes much of the styling process. It does this by including much of the common styling choices as an optional pre-built class that can be attached to HTML elements, most commonly divs or spans. In a typical workflow, you would need to create the CSS for these class yourself in a .css file and import those styles. Over a large project, you might begin to realize that the styles can get messy and/or redundnant. Using Bootstrap's built-in styles would instead allow you to rely on their basic foundational stylistic building blocks, requiring less custom CSS from you.
```

Comment:
What is the difference between css and html? what are the benefits of using bosstrap instead of other framework? why do we use those?

Reply:
I think my guide is more for people who at least know what CSS and HTML are or can do, so I believe it might be a little unnecessary to include something like the difference between CSS and HTML I added in the preface that it is more intended for someone who can make a simple web page, and I believe this guide will only prove useful to someone who is at that level.

However, I do agree that I should put some information about other frameworks in here.

Reply:
That makes sense. Thank you for the clarification!!!